php: simplify server configuration

vanyauhalin opened the pull request:
Pending #424, but is ready to be tested, if #424 isn't in question.

rivexe reviewed (commented):
1. Almost all server responses seem to give `404` now. This about `editor` root, `upload` root, and others. I don't know, maybe it's related to the server configuration, but it's strange if I need any additional manipulations with this, because before it was enough to run `Apache` / `IIS` and not get any problems.

2. Secondly, I didn’t understand the moment with [this header](https://github.com/ONLYOFFICE/document-server-integration/pull/425/commits/1898da660e8dce261cf2d7faec812a5ad6f6db6f#diff-839b5e2067468733913e390043a0c10672117ab2d4f210d251fe0f1fddd76025R53).

3. This applies **_not only to this_** PR, but also to previous ones. It looks like you've added some **_new dependencies_**. Shouldn't they be added to the **_license files_**? And you need to check which licenses these packages have.

vanyauhalin commented:
> * Almost all server responses seem to give `404` now. This about `editor` root, `upload` root, and others. I don't know, maybe it's related to the server configuration, but it's strange if I need any additional manipulations with this, because before it was enough to run `Apache` / `IIS` and not get any problems.
> * Secondly, I didn’t understand the moment with [this header](https://github.com/ONLYOFFICE/document-server-integration/pull/425/commits/1898da660e8dce261cf2d7faec812a5ad6f6db6f#diff-839b5e2067468733913e390043a0c10672117ab2d4f210d251fe0f1fddd76025R53).
> * This applies **_not only to this_** PR, but also to previous ones. It looks like you've added some **_new dependencies_**. Shouldn't they be added to the **_license files_**? And you need to check which licenses these packages have.

Ye, I think you're right. The first one is most likely because I deleted the IIS config. In a future pull request, I added Docker with Nginx to replace IIS and make all projects have a consistent appearance. Now, I'd like to ask you to execute `make server-dev`, which should set up a development server on port 9000. If it doesn't work, then I'll restore the IIS file and include the commit for its removal in the next pull request.

That is regarding the header. I didn't fully understand it either, but it was already present in the previous server configuration[^1][^2], so I simply copied it as it's.

And in the last one, I misunderstood you because I didn't add any new dependencies here.

[^1]: https://github.com/ONLYOFFICE/document-server-integration/blob/1d4bc95c295147b17145a2792094c25cce1bc651/web/documentserver-example/php/ajax.php#L57
[^2]: https://github.com/ONLYOFFICE/document-server-integration/blob/1d4bc95c295147b17145a2792094c25cce1bc651/web/documentserver-example/php/webeditor-ajax.php#L61

rivexe commented:
> > * Almost all server responses seem to give `404` now. This about `editor` root, `upload` root, and others. I don't know, maybe it's related to the server configuration, but it's strange if I need any additional manipulations with this, because before it was enough to run `Apache` / `IIS` and not get any problems.
> > * Secondly, I didn’t understand the moment with [this header](https://github.com/ONLYOFFICE/document-server-integration/pull/425/commits/1898da660e8dce261cf2d7faec812a5ad6f6db6f#diff-839b5e2067468733913e390043a0c10672117ab2d4f210d251fe0f1fddd76025R53).
> > * This applies **_not only to this_** PR, but also to previous ones. It looks like you've added some **_new dependencies_**. Shouldn't they be added to the **_license files_**? And you need to check which licenses these packages have.
> 
> Ye, I think you're right. The first one is most likely because I deleted the IIS config. In a future pull request, I added Docker with Nginx to replace IIS and make all projects have a consistent appearance. Now, I'd like to ask you to execute `make server-dev`, which should set up a development server on port 9000. If it doesn't work, then I'll restore the IIS file and include the commit for its removal in the next pull request.
> 
> That is regarding the header. I didn't fully understand it either, but it was already present in the previous server configuration[1](#user-content-fn-1-79e1ae2b399932c267d6833c7051643d)[2](#user-content-fn-2-79e1ae2b399932c267d6833c7051643d), so I simply copied it as it's.
> 
> And in the last one, I misunderstood you because I didn't add any new dependencies here.
> 
> ## Footnotes
> 1. https://github.com/ONLYOFFICE/document-server-integration/blob/1d4bc95c295147b17145a2792094c25cce1bc651/web/documentserver-example/php/ajax.php#L57 [↩](#user-content-fnref-1-79e1ae2b399932c267d6833c7051643d)
> 2. https://github.com/ONLYOFFICE/document-server-integration/blob/1d4bc95c295147b17145a2792094c25cce1bc651/web/documentserver-example/php/webeditor-ajax.php#L61 [↩](#user-content-fnref-2-79e1ae2b399932c267d6833c7051643d)

1. Yes, using `make` , the server starts on port `9000` , but there is probably still a question of the availability of a deployment in different ways. I really like how you made it easy to work with deployment in the examples, but most of the examples are not for our development, but for other developers. Not everyone works with Docker, it is more convenient for someone to deploy on an already configured server, etc. Here we need to think about whether it is worth getting rid of old configs.

2. Apparently I didn't notice it before =D

3. In this PR - no, but apparently somewhere earlier. I just saw that there is a lot of new stuff in `/vendor` (besides `JWT` library, `linter` and `PhpUnit`).
